### PR TITLE
INFENG-6986 update docker image and use nobody user in dockerfile

### PR DESCRIPTION
**Pull request by blakeroberts-wk**
### Story:
We need to give aviary what he ... err ... she ... wants!

### Acceptance Criteria:
- [x] At least one InfRe Squad 2 member has reviewed and +1'd
- [x] Code has been tested and results documented
- [x] Unit tests have been updated
- [x] Updates to documentation if necessary
- [x] Verify and document changes to any other Messaging components
- [x] Pull request made against the 'develop' branch, not master

### Design Notes:
I went with a stateless pub/sub server/client dependency injection model to refactor the docker scratch image to support the nobody *nix user.

### How To Test:
```bash
docker run -it drydock.workiva.net/workiva/frugal:1575160 --help
```

### My Test Results:
```
$ docker run -it drydock.workiva.net/workiva/frugal:1575160 --help
Unable to find image 'drydock.workiva.net/workiva/frugal:1575160' locally
1575160: Pulling from workiva/frugal
a217dc0ab271: Already exists
8532ff613d02: Already exists
Digest: sha256:5d311e1d546e92037062d29ee8369ccba4fdc86beaf3bd43cd065128ba75ec4c
Status: Downloaded newer image for drydock.workiva.net/workiva/frugal:1575160
NAME:
   frugal - a tool for code generation

USAGE:
   frugal [global options] [arguments...]

GLOBAL OPTIONS:
   --help, -h     show help
   --gen value    generate code with a registered generator and optional parameters (lang[:key1=val1[,key2[,key3=val3]]])
                      dart:
                          library_prefix:  Generate code that can be used within an existing library. Use a dot-separated string, e.g. "my_parent_lib.src.gen"
                          use_enums:       Generate enums as enums rather than a class with numerical constants
                          use_vendor:      Use specified import references for vendored includes and do not generate code for them
                      go:
                          async:           Generate async client code using channels
                          frugal_import:   Override Frugal package import path (default: github.com/Workiva/frugal/lib/go)
                          package_prefix:  Package prefix for generated files
                          slim:            Generate slim type definitions (WARNING: code generated by this may break code consumers, protocol logic should not change)
                          thrift_import:   Override Thrift package import path (default: git.apache.org/thrift.git/lib/go/thrift)
                          use_vendor:      Use specified import references for vendored includes and do not generate code for them
                      gopherjs:
                          package_prefix:  Package prefix for generated files
                          use_vendor:      Use specified import references for vendored includes and do not generate code for them
                      html:
                          standalone:  Self-contained mode, includes all CSS in the HTML files. Generates no style.css file, but HTML files will be larger
                      java:
                          async:                  Generate async client code using futures
                          boxed_primitives:       Generate primitives as the boxed equivalents
                          generated_annotations:  [undated|suppress] undated: suppress the date at @Generated annotations, suppress: suppress @Generated annotations entirely
                          use_vendor:             Use specified import references for vendored includes and do not generate code for them
                      py:
                          asyncio:         Generate code for use with asyncio (compatible with Python 3.5 or above)
                          package_prefix:  Package prefix for generated files
                          tornado:         Generate code for use with Tornado (compatible with Python 2.7)
   --out value    set the output location for generated files (no gen-* folder will be created)
   --delim value  set the delimiter for pub/sub topic tokens (default: ".")
   --recurse, -r  generate included files
   --verbose, -v  verbose mode
   --version      print the version
   --audit value  frugal file to run audit against
```

#### Reviewers:
@Workiva/product2

### CODE REVIEW

**Comment by aviary2-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.
<details>
<summary>(1) Image validation issues were detected</summary>
<li>Repo does not appear to install package updates in deployable image</li>
</details>

## Action Items
* Ensure [package updates](https://wiki.atl.workiva.net/display/SECURITY/Security+Requirements+for+Building+Containers#SecurityRequirementsforBuildingContainers-PackageUpdating) are being installed.
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://dev.workiva.net/docs/teams/information-security/aviary#configuring-aviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

**Comment by matthewlorimor-wf** [comment]
aviary override required

**Comment by matthewlorimor-wf** [comment]
Override was required due to ```FROM scratch``` and copy usage to build the container. Files copied into the scratch stage are not related to packages or updates in any way.

Security +1

**Comment by jeffreymelvin-wf** [comment]
@aviary-wf

**Comment by jeffreymelvin-wf** [comment]
@aviary-wf

File: Dockerfile [hunk 1]
```diff
@@ -35,6 +40,18 @@ ARG BUILD_ARTIFACTS_ARTIFACTORY=/go/src/github.com/Workiva/frugal/frugal-*.jar
 ARG BUILD_ARTIFACTS_PUB=/go/src/github.com/Workiva/frugal/frugal.pub.tgz
 ARG BUILD_ARTIFACTS_TEST_RESULTS=/go/src/github.com/Workiva/frugal/test_results/*
 
+RUN mkdir /veracode && \
+	tar czf /veracode/dart.tar.gz /go/src/github.com/Workiva/frugal/lib/dart && \
+	tar czf /veracode/go.tar.gz /go/src/github.com/Workiva/frugal/lib/go && \
```
Comment (yuanmwang-wf) [inline Dockerfile]: nit: `s/"/go/src/github.com/Workiva/frugal"/$FRUGAL_HOME` in these lines?

**Comment by yuanmwang-wf** [comment]
+1

File: Dockerfile (hunk 1 above)
Comment (blakeroberts-wk) [reply Dockerfile]: I would, but I think we will be removing this. veracode is a t1 repo thing, frugal is not. But aviary wanted it. I am talking with infosec atm and getting it straightened out.

**Comment by tylerrinnan-wf** [comment]
+1

**Comment by danielnovotny-wf** [comment]
+1

**Comment by stevenosborne-wf** [comment]
+1

**Comment by tylerrinnan-wf** [comment]
+1

**Comment by blakeroberts-wk** [comment]
rosie

**Comment by blakeroberts-wk** [comment]
@Workiva/release-management-p

**Pull request merged**